File: exchange-rate-webapp/src/app/core/interfaces/exchange-rate.ts
export interface ExchangeRateRequest {
  fromCurrency: string;
  toCurrency: string;
  amount: number;
}

export interface ExchangeRateResponse {
  fromCurrency: string;
  toCurrency: string;
  amount: number;
  conversionRate: number;
  conversionResult: number;
}

export interface ExchangeRateHistory {
  id: string;
  baseCurrency: string;
  targetCurrency: string;
  amountConvert: string;
  conversionRate: number;
  conversionResult: number;
  timestamp: Date;
}

File: exchange-rate-api/src/main/java/pe/interbank/exchange/domain/ExchangeRate.java
package pe.interbank.exchange.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "exchange_rate_history")
public class ExchangeRate extends AbstractAuditingEntity {

    @Id
    @Column(name = "id")
    private String id = UUID.randomUUID().toString();

    @Column(name = "base_currency", nullable = false)
    private String baseCurrency;

    @Column(name = "target_currency", nullable = false)
    private String targetCurrency;

    @Column(name = "amount_convert", nullable = false)
    private BigDecimal amountConvert;

    @Column(name = "conversion_rate", nullable = false)
    private BigDecimal conversionRate;

    @Column(name = "conversion_result", nullable = false)
    private BigDecimal conversionResult;

    @Column(name = "timestamp")
    private Instant timestamp = Instant.now();
}
